packages/datadog-esbuild/index.js

```diff
@@ -109,46 +110,48 @@ module.exports.setup = function (build) {
       // https://esbuild.github.io/plugins/#on-resolve-arguments
       return {
         path: fullPathToModule,
-        namespace: NAMESPACE,
         pluginData: {
           version: packageJson.version,
           pkg: extracted.pkg,
           path: extracted.path,
           full: fullPathToModule,
           raw: args.path,
+          pkgOfInterest: true,
           internal
         }
       }
-    } else if (args.namespace === NAMESPACE) {
-      // The datadog namespace is used when requiring files that are injected during the onLoad stage
-
-      if (builtins.has(args.path)) return
-
-      return {
-        path: require.resolve(args.path, { paths: [ args.resolveDir ] }),
-        namespace: 'file'
-      }
     }
   })
 
-  build.onLoad({ filter: /.*/, namespace: NAMESPACE }, args => {
+  build.onLoad({ filter: /.*/ }, args => {
+    if (!args.pluginData?.pkgOfInterest) {
+      return
+    }
+
     const data = args.pluginData
 
     if (DEBUG) console.log(`LOAD: ${data.pkg}@${data.version}, pkg "${data.path}"`)
 
-    const path = data.raw !== data.pkg
+    const pkgPath = data.raw !== data.pkg
       ? `${data.pkg}/${data.path}`
       : data.pkg
 
+    // Read the content of the module file of interest
+    const fileCode = fs.readFileSync(args.path, 'utf8')
+
     const contents = `
+      // Module code from ${args.path}
+      (function() {
+        ${fileCode}
+      })(...arguments);
```

Comment:
For future reviewers: This is for a weird case where a user can access `arguments` from the root of a commonJS file, which contains `[exports, require, module, __filename, __dirname]`. Surely nobody would use `arguments` directly but we've certainly seen weirder things happen.